pkg/controller/machinepool/machinepool_controller_test.go

```diff
@@ -380,8 +387,122 @@ func TestRemoteMachineSetReconcile(t *testing.T) {
 					ms.Spec.Template.Spec.ObjectMeta.Labels["test-label-1"] = "test-value-1"
 					ms.Spec.Template.Spec.ObjectMeta.Labels["test-label-2"] = ""
 					ms.Spec.Template.Spec.Taints = append(ms.Spec.Template.Spec.Taints, corev1.Taint{
-						Key:   "test-taint",
-						Value: "test-value",
+						Key:    "test-taint",
+						Value:  "test-value",
+						Effect: "NoSchedule",
+					})
+					return ms
+				}(),
+			},
+		},
```

Comment:
Can we please get a test where the machinepool has duplicates? I predict that we'll end up "ignoring" (actually overwriting) all but the last one. And I think I don't mind that behavior as much as leaving (identical) duplicates in the remote MachineSet...

Reply:
Done. Also ensured the expected entries are collapsed.